src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -295,9 +315,11 @@ public function searchAction(Request $request)
      * @param int $orderId
      */
     #[AdminSecurity("is_granted('read', request.get('_legacy_controller'))", redirectRoute: 'admin_orders_index')]
-    public function generateInvoicePdfAction($orderId)
-    {
-        $this->get('prestashop.adapter.pdf.order_invoice_pdf_generator')->generatePDF([$orderId]);
+    public function generateInvoicePdfAction(
+        int $orderId,
+        OrderInvoicePdfGenerator $invoicePdfGenerator,
+    ) {
+        $invoicePdfGenerator->generatePDF([$orderId]);
```

Comment:
Is it possible to do the same as for credit slip here? Generate a string and return a BinaryResponse instead?